x-pack/plugins/apm/public/components/shared/Stacktrace/stacktraceUtils.ts

```diff
@@ -7,35 +7,35 @@
 import { get, isEmpty } from 'lodash';
 import { Stackframe } from '../../../../typings/APMDoc';
 
-export interface StackframeCollapsed extends Stackframe {
-  libraryFrame?: boolean;
-  stackframes?: Stackframe[];
+interface StackframesGroup {
+  isLibraryFrame: boolean;
+  stackframes: Stackframe[];
 }
```

Comment:
The previous return type was an array of elements: either a stackframe, or a nested sub-list of stackframes. Example: 
```
[
    Stackframe,
    {libraryFrame, stackframes: [...]},
    Stackframe,
    Stackframe,
    {libraryFrame, stackframes: [...]},
    ...
]
```
To simplify the function, the new return type is always an array of `StackframesGroup` types (groups of size 1 are valid). 

Ultimately, the render function is responsible for rendering the `StackframesGroup` data as either a collapsed set of stackframes or an expanded stackframe (with source, context, styles, etc.)

Reply:
Great simplification!